File: FSharp/Api/Data/ApplicationDbContext.fs
namespace Api.Data

open Api.Data.Configuration
open Common.Entities
open Microsoft.AspNetCore.Identity.EntityFrameworkCore
open Microsoft.EntityFrameworkCore

type ApplicationDbContext(options: DbContextOptions<ApplicationDbContext>) =
  inherit IdentityDbContext<ApplicationUser>(options)
  member this.ApplicationRoles = base.Set<ApplicationRole>()
  member this.ApplicationUsers = base.Set<ApplicationUser>()
  member this.ApplicationUserRoles = base.Set<ApplicationUserRole>()

  override this.OnModelCreating(builder: ModelBuilder) =
    base.OnModelCreating(builder)

    builder
      .ApplyConfiguration(ApplicationUserConfiguration())
      .ApplyConfiguration(ApplicationRoleConfiguration())
      .ApplyConfiguration(ApplicationUserRoleConfiguration())
    |> ignore

File: FSharp/DataContext/Data/ApplicationDbContext.cs
using Common.Entities;
using DataContext.Data.Configuration;
using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore;

namespace DataContext.Data;

public class ApplicationDbContext(DbContextOptions<ApplicationDbContext> options)
    : IdentityDbContext<ApplicationUser>(options)
{
    public DbSet<ApplicationRole> ApplicationRoles => Set<ApplicationRole>();
    public DbSet<ApplicationUser> ApplicationUsers => Set<ApplicationUser>();
    public DbSet<ApplicationUserRole> ApplicationUserRoles => Set<ApplicationUserRole>();

    protected override void OnModelCreating(ModelBuilder builder)
    {
        base.OnModelCreating(builder);
        builder
            .ApplyConfiguration(new ApplicationUserConfiguration())
            .ApplyConfiguration(new ApplicationRoleConfiguration())
            .ApplyConfiguration(new ApplicationUserRoleConfiguration());
    }
}
